backend/inspirehep/records/tasks.py

```diff
@@ -135,3 +142,17 @@ def regenerate_author_records_table_entries(uuids_to_regenerate):
         record.update_authors_records_table()
         record.update_self_citations()
         db.session.commit()
+
+
+def remove_duplicate_refs_from_record(matched_inspire_record, referenced_records_in_path, path):
+    affiliations_path = ".".join(path.split(".")[:-2])
+    affiliations = flatten_list(get_value(matched_inspire_record, affiliations_path, []))
+
+    if len(affiliations) == len(referenced_records_in_path):
+        return None
+
+    for affiliation in affiliations:
```

Comment:
So now you are iterating over the list of dicts in, let's say, publication_info and you are removing items from it if they're already on the list. Why it can't be done with dedupe list of dicts instead of this whole function?